### PR TITLE
Student records

### PR DESCRIPTION
**Pull request by jaideepbhargava**
Added an Api driven student records managing system

### CODE REVIEW

File: Day3/exercise2/Controllers/Student.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+	if err != nil {
+		fmt.Println(err.Error())
+		context.AbortWithStatus(http.StatusNotFound)
+	} else {
+		context.JSON(http.StatusOK, record)
+	}
+}
+
+func GetUserByID(c *gin.Context) {
```
Comment (Sampriti-Mitra) [inline Day3/exercise2/Controllers/Student.go:36]: GetStudentById

File: Day3/exercise2/Controllers/Student.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	var user Models.Student
+	err := Models.GetRecordsByID(&user, id)
+	if err != nil {
+		c.AbortWithStatus(http.StatusNotFound)
+	} else {
+		c.JSON(http.StatusOK, user)
+	}
+}
+
+func UpdateUser(c *gin.Context) {
```
Comment (Sampriti-Mitra) [inline Day3/exercise2/Controllers/Student.go:47]: Naming

File: Day3/exercise2/Controllers/Student.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+func CreateStudent(context *gin.Context){
+	var record Models.Student
+	context.BindJSON(&record)
+	err := Models.AddRecords(&record)
+
+	if err != nil {
+		fmt.Println(err.Error())
+		context.AbortWithStatus(http.StatusNotFound)
```
Comment (Sampriti-Mitra) [inline Day3/exercise2/Controllers/Student.go:30]: error status should be different, should be according to error, not a generic one

File: Day3/exercise2/Models/StudentModel.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+	Id int `json:"id"`
+	Name string `json:"name"`
+	LastName string `json:"last_name"`
+	DOB string `json:"dob"`
+	Address string `json:"address"`
+	Subject string `json:"subject"`
+	Marks int `json:"marks"`
+}
+
+func (student *Student) StudentTable() string{
```
Comment (jaideepbhargava) [inline Day3/exercise2/Models/StudentModel.go:13]: Add TableName()

**Pull request merged**